// @primer/primitives/dist/build/platforms/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.typescript = exports.typeDefinitions = exports.styleLint = exports.json = exports.javascript = exports.figma = exports.fallbacks = exports.docJson = exports.deprecatedJson = exports.css = void 0;
var css_1 = require("./css");
Object.defineProperty(exports, "css", { enumerable: true, get: function () { return css_1.css; } });
var deprecatedJson_1 = require("./deprecatedJson");
Object.defineProperty(exports, "deprecatedJson", { enumerable: true, get: function () { return deprecatedJson_1.deprecatedJson; } });
var docJson_1 = require("./docJson");
Object.defineProperty(exports, "docJson", { enumerable: true, get: function () { return docJson_1.docJson; } });
var fallbacks_1 = require("./fallbacks");
Object.defineProperty(exports, "fallbacks", { enumerable: true, get: function () { return fallbacks_1.fallbacks; } });
var figma_1 = require("./figma");
Object.defineProperty(exports, "figma", { enumerable: true, get: function () { return figma_1.figma; } });
var javascript_1 = require("./javascript");
Object.defineProperty(exports, "javascript", { enumerable: true, get: function () { return javascript_1.javascript; } });
var json_1 = require("./json");
Object.defineProperty(exports, "json", { enumerable: true, get: function () { return json_1.json; } });
var styleLint_1 = require("./styleLint");
Object.defineProperty(exports, "styleLint", { enumerable: true, get: function () { return styleLint_1.styleLint; } });
var typeDefinitions_1 = require("./typeDefinitions");
Object.defineProperty(exports, "typeDefinitions", { enumerable: true, get: function () { return typeDefinitions_1.typeDefinitions; } });
var typescript_1 = require("./typescript");
Object.defineProperty(exports, "typescript", { enumerable: true, get: function () { return typescript_1.typescript; } });
