// @primer/primitives/dist/build/schemas/colorToken.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.colorToken = void 0;
const zod_1 = require("zod");
const colorHexValue_1 = require("./colorHexValue");
const referenceValue_1 = require("./referenceValue");
const alphaValue_1 = require("./alphaValue");
const baseToken_1 = require("./baseToken");
const collections_1 = require("./collections");
const scopes_1 = require("./scopes");
const tokenType_1 = require("./tokenType");
exports.colorToken = baseToken_1.baseToken
    .extend({
    $value: zod_1.z.union([colorHexValue_1.colorHexValue, referenceValue_1.referenceValue]),
    $type: (0, tokenType_1.tokenType)('color'),
    alpha: alphaValue_1.alphaValue.optional().nullable(),
    mix: zod_1.z
        .object({
        color: zod_1.z.string(),
        weight: zod_1.z.number().min(0).max(1),
    })
        .nullable()
        .optional(),
    $extensions: zod_1.z
        .object({
        'org.primer.figma': zod_1.z.object({
            collection: (0, collections_1.collection)([
                'base/color/light',
                'base/color/light-high-contrast',
                'base/color/dark',
                'base/color/dark-dimmed',
                'base/color/dark-high-contrast',
                'mode',
                'pattern/mode',
            ]).optional(),
            mode: (0, collections_1.mode)([
                'light',
                'dark',
                'dark dimmed',
                'light high contrast',
                'dark high contrast',
                'light colorblind',
                'dark colorblind',
                'light tritanopia',
                'dark tritanopia',
            ]).optional(),
            scopes: (0, scopes_1.scopes)(['all', 'bgColor', 'fgColor', 'borderColor', 'effectColor']).optional(),
        }),
    })
        .optional(),
})
    .strict();
