// @primer/primitives/dist/build/schemas/dimensionToken.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.dimensionToken = void 0;
const zod_1 = require("zod");
const baseToken_1 = require("./baseToken");
const referenceValue_1 = require("./referenceValue");
const collections_1 = require("./collections");
const scopes_1 = require("./scopes");
const dimensionValue_1 = require("./dimensionValue");
const tokenType_1 = require("./tokenType");
exports.dimensionToken = baseToken_1.baseToken
    .extend({
    $value: zod_1.z.union([dimensionValue_1.dimensionValue, referenceValue_1.referenceValue]),
    $type: (0, tokenType_1.tokenType)('dimension'),
    $extensions: zod_1.z
        .object({
        'org.primer.figma': zod_1.z.object({
            collection: (0, collections_1.collection)(['base/size', 'functional/size', 'pattern/size', 'typography']),
            scopes: (0, scopes_1.scopes)([
                'all',
                'size',
                'gap',
                'radius',
                'borderColor',
                'effectFloat',
                'fontSize',
                'letterSpacing',
                'lineHeight',
                'paragraphSpacing',
                'paragraphIndent',
            ]),
        }),
    })
        .optional(),
})
    .strict();
