// @primer/primitives/dist/build/schemas/numberToken.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.numberToken = void 0;
const zod_1 = require("zod");
const baseToken_1 = require("./baseToken");
const referenceValue_1 = require("./referenceValue");
const tokenType_1 = require("./tokenType");
const scopes_1 = require("./scopes");
const collections_1 = require("./collections");
exports.numberToken = baseToken_1.baseToken
    .extend({
    $value: zod_1.z.union([zod_1.z.number(), referenceValue_1.referenceValue]),
    $type: (0, tokenType_1.tokenType)('number'),
    $extensions: zod_1.z
        .object({
        'org.primer.data': zod_1.z
            .object({
            fontSize: zod_1.z.number().optional(),
        })
            .optional(),
        'org.primer.figma': zod_1.z
            .object({
            collection: (0, collections_1.collection)(['typography']).optional(),
            scopes: (0, scopes_1.scopes)(['fontWeight', 'lineHeight']).optional(),
        })
            .optional(),
    })
        .optional(),
})
    .strict();
