// @primer/primitives/dist/build/schemas/typographyToken.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.typographyToken = exports.typographyValue = void 0;
const zod_1 = require("zod");
const referenceValue_1 = require("./referenceValue");
const dimensionValue_1 = require("./dimensionValue");
const baseToken_1 = require("./baseToken");
const fontWeightValue_1 = require("./fontWeightValue");
const tokenType_1 = require("./tokenType");
exports.typographyValue = zod_1.z.object({
    fontSize: zod_1.z.union([dimensionValue_1.dimensionValue, referenceValue_1.referenceValue]),
    lineHeight: zod_1.z.union([dimensionValue_1.dimensionValue, referenceValue_1.referenceValue]).optional(),
    fontWeight: zod_1.z.union([fontWeightValue_1.fontWeightValue, referenceValue_1.referenceValue]),
    fontFamily: zod_1.z.union([zod_1.z.string().min(1), referenceValue_1.referenceValue]),
});
exports.typographyToken = baseToken_1.baseToken
    .extend({
    $value: zod_1.z.union([exports.typographyValue, referenceValue_1.referenceValue]),
    $type: (0, tokenType_1.tokenType)('typography'),
})
    .strict();
