// @primer/primitives/dist/build/filters/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.isTypography = exports.isSource = exports.isShadow = exports.isNumber = exports.isFontWeight = exports.isFontFamily = exports.isFromFile = exports.isDeprecated = exports.isDimension = exports.isColorWithMix = exports.isColorWithAlpha = exports.isColor = exports.isBorder = void 0;
var isBorder_1 = require("./isBorder");
Object.defineProperty(exports, "isBorder", { enumerable: true, get: function () { return isBorder_1.isBorder; } });
var isColor_1 = require("./isColor");
Object.defineProperty(exports, "isColor", { enumerable: true, get: function () { return isColor_1.isColor; } });
var isColorWithAlpha_1 = require("./isColorWithAlpha");
Object.defineProperty(exports, "isColorWithAlpha", { enumerable: true, get: function () { return isColorWithAlpha_1.isColorWithAlpha; } });
var isColorWithMix_1 = require("./isColorWithMix");
Object.defineProperty(exports, "isColorWithMix", { enumerable: true, get: function () { return isColorWithMix_1.isColorWithMix; } });
var isDimension_1 = require("./isDimension");
Object.defineProperty(exports, "isDimension", { enumerable: true, get: function () { return isDimension_1.isDimension; } });
var isDeprecated_1 = require("./isDeprecated");
Object.defineProperty(exports, "isDeprecated", { enumerable: true, get: function () { return isDeprecated_1.isDeprecated; } });
var isFromFile_1 = require("./isFromFile");
Object.defineProperty(exports, "isFromFile", { enumerable: true, get: function () { return isFromFile_1.isFromFile; } });
var isFontFamily_1 = require("./isFontFamily");
Object.defineProperty(exports, "isFontFamily", { enumerable: true, get: function () { return isFontFamily_1.isFontFamily; } });
var isFontWeight_1 = require("./isFontWeight");
Object.defineProperty(exports, "isFontWeight", { enumerable: true, get: function () { return isFontWeight_1.isFontWeight; } });
var isNumber_1 = require("./isNumber");
Object.defineProperty(exports, "isNumber", { enumerable: true, get: function () { return isNumber_1.isNumber; } });
var isShadow_1 = require("./isShadow");
Object.defineProperty(exports, "isShadow", { enumerable: true, get: function () { return isShadow_1.isShadow; } });
var isSource_1 = require("./isSource");
Object.defineProperty(exports, "isSource", { enumerable: true, get: function () { return isSource_1.isSource; } });
var isTypography_1 = require("./isTypography");
Object.defineProperty(exports, "isTypography", { enumerable: true, get: function () { return isTypography_1.isTypography; } });
