// @primer/primitives/dist/build/transformers/fontFamilyToCss.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.fontFamilyToCss = exports.parseFontFamily = void 0;
const filters_1 = require("../filters");
const getTokenValue_1 = require("./utilities/getTokenValue");
const hasSpaceInString_1 = require("./utilities/hasSpaceInString");
/**
 * takes a value and returns it if its a string or concats strings in an array quoting strings with spaces
 * @param value
 * @returns string
 */
const parseFontFamily = (value) => {
    if (typeof value === 'string') {
        return value;
    }
    if (Array.isArray(value)) {
        return value
            .map((string) => {
            if (typeof string !== 'string') {
                throw new Error(`Invalid value in array ${string}, must be a string`);
            }
            return (0, hasSpaceInString_1.hasSpaceInString)(string) ? `'${string}'` : string;
        })
            .join(', ');
    }
    throw new Error(`Invalid value ${value}, should be a string or array of strings`);
};
exports.parseFontFamily = parseFontFamily;
/**
 * @description converts fontFamily tokens value to string
 * @type value transformer — [StyleDictionary.ValueTransform](https://github.com/amzn/style-dictionary/blob/main/types/Transform.d.ts)
 * @matcher matches all tokens of $type `fontFamily`
 * @transformer returns a string
 */
exports.fontFamilyToCss = {
    type: `value`,
    transitive: true,
    matcher: filters_1.isFontFamily,
    transformer: (token) => (0, exports.parseFontFamily)((0, getTokenValue_1.getTokenValue)(token)),
};
